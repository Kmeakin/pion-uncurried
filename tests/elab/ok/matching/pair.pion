enum Pair(A: Type, B: Type) {
    MkPair(x: A, y: B),
}

let fst = fn(A: Type, B: Type, p: Pair(A, B)) => match p {
    MkPair(x, _) => x,
};

let snd = fn(A: Type, B: Type, p: Pair(A, B)) => match p {
    MkPair(_, y) => y,
};
