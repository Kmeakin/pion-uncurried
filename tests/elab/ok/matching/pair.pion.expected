enum Pair(A: Type, B: Type): Type {
    MkPair(x: A, y: B): Pair(A, B),
}

let fst: fn(A: Type, B: Type, p: Pair(A, B)) -> A = fn(A: Type, B: Type, p: Pair(A, B)) => match p {
    MkPair(x, _) => x,
};

let snd: fn(A: Type, B: Type, p: Pair(A, B)) -> B = fn(A: Type, B: Type, p: Pair(A, B)) => match p {
    MkPair(_, y) => y,
};
